frontend/src/components/widgets/DataFrame/hooks/useDataEditor.ts

```diff
@@ -149,7 +152,8 @@ function useDataEditor(
             col++
           ) {
             const column = columns[col]
-            if (column.isEditable) {
+            // Only allow deletion if the column is editable and not configured as required
+            if (column.isEditable === true && column.isRequired !== true) {
```

Comment:
Personally, I think
```if (column.isEditable && !column.isRequired) {``` 
is a lot cleaner and makes it so you dont need the comment.

Reply:
Ah I totally forgot about the undefined and null checks. If we were to do that, I would expect a comment about that. I'll leave it up to you which one you wanna do.

Reply:
Changed it